### PR TITLE
refactor(Modal): Separate Modal into Dialog component and createModal

### PR DESCRIPTION
**Pull request by u9520107**
1. Refactor the Dialog Component out of Modal Component
2. Implement a createModal function to wrap anything into a Modal

### CODE REVIEW

File: packages/ringcentral-widgets/components/Dialog/index.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import styles from './styles.scss';
+import i18n from './i18n';
+import Button from '../Button';
+import CloseIcon from '../../assets/images/CloseIcon.svg';
+
+function FlatButton({
+  className, disabled, onClick, children
+}) {
+  return (
+    <div
```
Comment (puresmash) [inline packages/ringcentral-widgets/components/Dialog/index.js]: May I ask why not using <button /> tag here?

File: packages/ringcentral-widgets/components/Dialog/index.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            {textConfirm || i18n.getString('confirm', currentLocale)}
+          </FlatButton>
+        ) : null}
+      </div>
+    );
+  return (
+    <div
+      className={classnames(styles.dialog, className)}
+    >
+      {showTitle ? (
```
Comment (puresmash) [inline packages/ringcentral-widgets/components/Dialog/index.js]: Is it better to add a container to wrap title and close button together?

File: packages/ringcentral-widgets/lib/createModal/index.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        modalClassName,
+        show,
+        onCancel,
+        clickOutToClose,
+        ...props
+      } = this.props;
+      const onClick = clickOutToClose ? onCancel : () => {};
+      return (
+        <div className={show ? classnames(styles.container, className) : styles.containerHidden}>
+          <div
```
Comment (puresmash) [inline packages/ringcentral-widgets/lib/createModal/index.js]: Could we extract a mask component?

File: packages/ringcentral-widgets/components/Dialog/index.js (hunk 2 above)
Comment (u9520107) [reply packages/ringcentral-widgets/components/Dialog/index.js]: Similarly, I think it's better to refactor the Dialog later, this PR is mainly to abstract the Modal design.

File: packages/ringcentral-widgets/components/Dialog/index.js (hunk 1 above)
Comment (u9520107) [reply packages/ringcentral-widgets/components/Dialog/index.js]: I only separated the original Dialog implementation from the Modal logic without changing the Dialog component itself to ensure backward compatibility. I'd suggest creating a new Dialog component with further refactoring and redesign later.

File: packages/ringcentral-widgets/components/Dialog/index.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        <div className={classnames(styles.header, headerClassName)}>
```
Comment (puresmash) [inline packages/ringcentral-widgets/components/Dialog/index.js]: Could we make header more semantic, prepare for the a11y. I think use section and header is appropriate here.

File: packages/ringcentral-widgets/components/Dialog/index.js (hunk 1 above)
Comment (puresmash) [reply packages/ringcentral-widgets/components/Dialog/index.js]: OK, I got the main idea.

**Pull request merged**